internal/charmstore/store.go

```diff
@@ -570,9 +570,13 @@ func (s *Store) findEntityInChannel(url *charm.URL, ch params.Channel, fields ma
 	}
 	var entityURL *charm.URL
 	if url.Series == "" {
-		for _, u := range baseEntity.ChannelEntities[ch] {
-			if entityURL == nil || seriesScore[u.Series] > seriesScore[entityURL.Series] {
+		var entitySeries string
+		for s, u := range baseEntity.ChannelEntities[ch] {
+			if entityURL == nil ||
+				u.Series == entityURL.Series && seriesScore[s] > seriesScore[entitySeries] ||
+				seriesScore[u.Series] > seriesScore[entityURL.Series] {
```

Comment:
Perhaps put this clause before the previous one as that
gets the most obvious conditions out of the way first?